src/GameStateMenu.cpp

```diff
@@ -10,143 +10,509 @@
+*/
 void GameStateMenu::sync() {
 
 }
 
+/**
+* Function: handle
+*
+* Date:
+* JF: February 8, 2017: added handler for window resizing event
+*
+* Designer:
+* Jacob Frank
+*
+* Programmer:
+* Jacob Frank
+*
+* Interface: handle()
+*
+* Returns: void
+*
+* Notes:
+* Function handles events occuring within the window to control
+* functionality of the main menu
+*/
 void GameStateMenu::handle() {
-    //Handle events on queue
-    while ( SDL_PollEvent( &event )) {
-        game.window.handleEvent(event);
-           switch( event.type ) {
-          case SDL_KEYDOWN:
-            switch( event.key.keysym.sym ) {
-            case SDLK_ESCAPE:
+	int x,y;
+	SDL_Keycode keyCode;
+
+	//Handle events on queue
+	SDL_WaitEvent( &event );
+	game.window.handleEvent(event);
+		switch ( event.type ) {
+      	case SDL_KEYDOWN:
+            keyCode = event.key.keysym.sym;
+            if (keyCode == SDLK_ESCAPE) {
                 play = false;
                 break;
-            default:
-                break;
             }
-            break;
-          case SDL_KEYUP:
-               switch( event.key.keysym.sym ) {
-            default:
-                   break;
+
+            for (size_t i = 0; i < NUM_TEXT_FIELDS; i++) {
+                if (activeTextbox[i]) {
+
+                    //Handling backspace
+                    if (keyCode == SDLK_BACKSPACE && textInput[i].length() > 0) {
+                        textInput[i].pop_back();
+                    }
+
+                    //Handle copy
+                    else if (keyCode == SDLK_c && SDL_GetModState() & KMOD_CTRL) {
+                        SDL_SetClipboardText( textInput[i].c_str() );
+                    }
+
+                    //Handle paste
+                    else if (keyCode == SDLK_v && SDL_GetModState() & KMOD_CTRL) {
+                        textInput[i] = SDL_GetClipboardText();
+                    }
+                }
+            }
+        	break;
+        case SDL_TEXTINPUT:
+            for (size_t i = 0; i < NUM_TEXT_FIELDS; i++) {
+                if (activeTextbox[i]) {
+                    //Ensures that a copy or paste is not occurring
+                    if (!((event.text.text[ 0 ] == 'c' || event.text.text[ 0 ] == 'C') &&
+                          (event.text.text[ 0 ] == 'v' || event.text.text[ 0 ] == 'V') &&
+                           SDL_GetModState() & KMOD_CTRL)) {
+                        //Update the string int displayed int he textbox
+                        if (textInput[i].length() < maxLength) {
+                            textInput[i] += event.text.text;
+                        }
+                    }
+                }
             }
             break;
-        case SDL_QUIT:
-            play = false;
-            break;
-          default:
-            break;
-        }
-    }
+      	case SDL_KEYUP: //Do nothing on key release
+       		switch ( event.key.keysym.sym ) {
```

Comment:
If keyup is ignored then there is no need for another switch inside
this should instead just be
```
case SDL_KEYUP: //Do nothing on key release
    break;
```